### ISSUE TITLE
[Grid - Basic] Menu button variant has empty table header

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2881]
### Provide a general summary of the issue here

Menu button variants have a menu button with actions in the last column, but the column is missing a descriptive header. This makes the function, and the structure of the table hard to interpret without exploration.

<img width="603" height="365" alt="Table header cell, which is the last element in the table header row is empty." src="https://github.com/user-attachments/assets/fd33b7db-56e3-40b5-b5e5-83cadebac8b8" />

### 🤔 Expected Behavior?

Table header has a text, which describes the purpose of the column.

### 😯 Current Behavior

Table header is empty and is not communicated to assistive technologies.

### 💁 Possible Solution

Add a text (e.g., only visible to screen reader users) which describes the purpose of the column. E.g., add a sr-only text "Actions" to explain the purpose of the column.

### 🖥️ Steps to Reproduce

Navigate to the header cell for the column with the functions -> Screen reader either reads "empty" or "space" (along with mentioning the table rows/columns and announcing that the cell is a table header).

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
